querydsl-core/src/main/java/com/querydsl/core/types/PredicateTemplate.java

```diff
@@ -32,7 +31,7 @@
     @Nullable
     private transient volatile Predicate not;
 
-    protected PredicateTemplate(Template template, ImmutableList<?> args) {
+    protected PredicateTemplate(Template template, List<?> args) {
         super(Boolean.class, template, args);
```

Comment:
```suggestion
        super(Boolean.class, template, Collections.unmodifiableList(new ArrayList<>(args)));
```

Reply:
Currently we use `ImmutableList` as an efficient way to ensure immutability of expression types. Using a plain `List`, we can only ensure this by making an unmodifiyable copy (to guard against copies of the original as well). This takes a performance penalty.